snakemake_executor_plugin_slurm/utils.py

```
@@ -40,3 +45,65 @@
+        gres = job.resources.gres
+        if not gres_re.match(gres):
+            raise WorkflowError(
+                f"Invalid GRES format: {gres}. Expected format: "
+                "'<name>:<number>' or '<name>:<type>:<number>' "
+                "(e.g., 'gpu:1' or 'gpu:tesla:2')"
+            )
+        gres_string = f" --gres={job.resources.gres}"
+    if job.resources.get("gpu"):
+        # ensure that gres is not set, if gpu and gpu_model are set
+        if job.resources.get("gres"):
+            raise WorkflowError("GRES and GPU are set. Please only set one of them.")
```

Ruff (0.8.2)

Undefined name `WorkflowError`

(F821)